src/tree.rs

```diff
@@ -26,6 +28,802 @@ pub struct JellyfishMerkleTree<'a, R> {
+                    } else {
+                        bail!(
+                            "Non-root null node exists with node key {:?}",
+                            next_node_key
+                        );
+                    }
+                }
+            }
+        }
+        bail!("Jellyfish Merkle tree has cyclic graph inside.");
+    }
+
+    /// Gets the proof that shows a list of keys up to `rightmost_key_to_prove` exist at `version`.
+    pub async fn get_range_proof(
+        &self,
+        rightmost_key_to_prove: KeyHash,
+        version: Version,
+    ) -> Result<SparseMerkleRangeProof> {
+        let (account, proof) = self.get_with_proof(rightmost_key_to_prove, version).await?;
+        ensure!(account.is_some(), "rightmost_key_to_prove must exist.");
+
+        let siblings = proof
+            .siblings()
+            .iter()
+            .rev()
+            .zip(rightmost_key_to_prove.0.iter_bits())
+            .filter_map(|(sibling, bit)| {
+                // We only need to keep the siblings on the right.
+                if !bit {
+                    Some(*sibling)
+                } else {
+                    None
+                }
+            })
+            .rev()
+            .collect();
+        Ok(SparseMerkleRangeProof::new(siblings))
+    }
+
+    pub async fn get(&self, key: KeyHash, version: Version) -> Result<Option<OwnedValue>> {
+        Ok(self.get_with_proof(key, version).await?.0)
+    }
+
+    async fn get_root_node(&self, version: Version) -> Result<Node> {
+        self.get_root_node_option(version)
+            .await?
+            .ok_or_else(|| format_err!("Root node not found for version {}.", version))
+    }
+
+    async fn get_root_node_option(&self, version: Version) -> Result<Option<Node>> {
+        let root_node_key = NodeKey::new_empty_path(version);
+        self.reader.get_node_option(&root_node_key).await
+    }
+
+    pub async fn get_root_hash(&self, version: Version) -> Result<RootHash> {
+        self.get_root_node(version)
+            .await
+            .map(|n| RootHash(n.hash()))
+    }
+
+    pub async fn get_root_hash_option(&self, version: Version) -> Result<Option<RootHash>> {
+        Ok(self
+            .get_root_node_option(version)
+            .await?
+            .map(|n| RootHash(n.hash())))
+    }
+
+    pub async fn get_leaf_count(&self, version: Version) -> Result<Option<usize>> {
+        if self.leaf_count_migration {
+            self.get_root_node(version).await.map(|n| n.leaf_count())
+        } else {
+            // When all children of an internal node are leaves, the leaf count is accessible
+            // even if the migration haven't started. In fact, in such a case, there's no difference
+            // in the old and new serialization format. Forcing it None here just to make the tests
+            // straightforward.
+            Ok(None)
+        }
+    }
+}
+
+/*
```

Comment:
Looks like the old implementation was commented out but not replaced, could we remove it so that we get the diff of old vs new?

Reply:
Yep, this was a miss & has been done